carapace-server/src/main/java/org/carapaceproxy/server/RequestHandler.java

```diff
@@ -558,6 +558,12 @@ public void receivedFromRemote(HttpObject msg, EndpointConnection connection) {
                 cleanResponseForCachedData(httpMessage);
             }
         }
+
+        // Custom response Headers
+        if (msg instanceof HttpResponse && action != null && action.responseHeaders != null) {
+            ((HttpResponse) msg).headers().add(action.responseHeaders);
```

Comment:
a question:
should we clone the HttpResponse object ?
are we serving it from the cache ?